### DISCUSSION TITLE
entropy

### DISCUSSION THREAD

**Discussion by giscus** [Jollywatt/jollywatt.github.io#1] [Posts]
# entropy

The formula for information entropy

http://localhost:4000/entropy

**Comment by Jollywatt** [comment]
> The only increasing functions $U(n)$ satisfying $U(nm) = U(n) + U(m)$ are multiples of $\log n$.

In case this claim in step 3 relies too heavily on theorems from analysis, here's a direct proof. It is sufficient to show that

$$ \frac{U(n)}{U(2)} = \frac{\log n}{\log 2} $$

We can prove this by contradiction. Assume the equality is instead $<$ or $>$. I will write the proof for both assumptions at once by writing $≶$, but of course we only make the assumptions one at a time.

Since they are not equal, there must be a rational number lying between the left and right hand sides:

$$ \frac{U(n)}{U(2)} ≶ \frac{p}{q} ≶ \frac{\log n}{\log 2}. $$

The left hand inequality implies $qU(n) ≶ pU(2)$, which by the additive property implies $U(n^q) ≶ U(2^p)$. (Iteratively apply $U(nm) = U(n) + U(m)$ to get $U(n^p) = pU(n)$.)

However, the right hand inequality implies $p\log 2 ≶ q\log n$, or exponentiating, $2^p ≶ n^q$. Since $U(n)$ is increasing, this implies $U(2^p) ≶ U(n^q)$ which directly contradicts the left hand side's implication.

Therefore the assumption holds, and $U(n) = K\log n$ for some $K = U(2)/\log 2$.
